features/notifications/utils/onInteractWithNotification.ts

```diff
@@ -51,18 +53,18 @@ export const onInteractWithNotification = (
   const conversationTopic = notificationData["contentTopic"] as
     | string
     | undefined;
-  const account =
-    notificationData["account"] || useAccountsStore.getState().currentAccount;
 
   if (conversationTopic) {
+    const account =
+      notificationData["account"] || useAccountsStore.getState().currentAccount;
+
+    // Fetch the conversation list to ensure we have the latest conversation list
+    // before navigating to the conversation
+    await fetchPersistedConversationListQuery(account);
     useAccountsStore.getState().setCurrentAccount(account, false);
-    const conversations = getChatStore(account).getState().conversations;
 
-    if (conversations[conversationTopic]) {
-      navigateToTopic(conversationTopic as ConversationTopic);
-    } else {
-      // App was probably not loaded!
-      setTopicToNavigateTo(conversationTopic);
-    }
+    navigateToTopic(conversationTopic as ConversationTopic);
+    setTopicToNavigateTo(undefined);
```

Comment:
Is this doing anything any more?

Reply:
I think this lingering around was responsible for the bug where you'd come back into the app and it'd try and deep link you again, so yes!